### ISSUE TITLE
[PATCH] base.pm not strict enough about nonexistent modules

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13453]
Migrated from [rt.perl.org#120685](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=120685) (status was 'resolved')

Searchable as RT120685$

**Comment by p5pRT** [comment]
### From @haarg

This is a bug report for perl from haarg@&#8203;haarg\.org\,
generated with the help of perlbug 1\.39 running under perl 5\.18\.1\.




---
When attempting to load a base module\, base\.pm ignores nonexistent
modules\.  The check it uses isn't very strict though\, and can end
up ignoring failures that it should have caught\.  Any error message
about failing to load a module that is in an eval will be ignored\.
Instead\, it should only ignore the specific module that it is
loading\.

Patch with test attached\.




---


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=low
&nbsp;    module=base


---
Site configuration information for perl 5\.18\.1&#8203;:

Configured by gknop at Mon Nov 11 14&#8203;:01&#8203;:12 EST 2013\.

Summary of my perl5 \(revision 5 version 18 subversion 1\) configuration&#8203;:
&nbsp;  Commit id&#8203;: 75c3005ee5b6ece3ad3c827cb300945f672c2bd1
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=13\.0\.2\, archname=darwin\-2level
&nbsp;    uname='darwin ophion 13\.0\.2 darwin kernel version 13\.0\.2&#8203;: sun sep
29 19&#8203;:38&#8203;:57 pdt 2013; root&#8203;:xnu\-2422\.75\.4~1release\_x86\_64 x86\_64 '
&nbsp;    config\_args='\-des \-Dusedevel \-Uversiononly
\-Dprefix=/Users/gknop/perl5/perls/v5\.18\.1'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=undef\, usemultiplicity=undef
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing
\-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-O3'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe
\-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 Compatible Apple LLVM 5\.0
\(clang\-500\.2\.79\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags ='
\-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-lgdbm \-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup
\-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:




---
@&#8203;INC for perl 5\.18\.1&#8203;:
&nbsp;    /Users/gknop/perl5/perls/v5\.18\.1/lib/site\_perl/5\.18\.1/darwin\-2level
&nbsp;    /Users/gknop/perl5/perls/v5\.18\.1/lib/site\_perl/5\.18\.1
&nbsp;    /Users/gknop/perl5/perls/v5\.18\.1/lib/5\.18\.1/darwin\-2level
&nbsp;    /Users/gknop/perl5/perls/v5\.18\.1/lib/5\.18\.1
&nbsp;    \.



---
Environment for perl 5\.18\.1&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/gknop
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/Users/gknop/perl5/perls/active/bin&#8203;:/usr/local/sbin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @haarg

<details><summary>0001-make-base.pm-more-strict-about-nonexistent-module-ch.patch</summary>

```diff
From 37fd83713e02d13fe29e66abc0925f3de7a52199 Mon Sep 17 00:00:00 2001
From: Graham Knop <haarg@haarg.org>
Date: Wed, 4 Dec 2013 05:54:02 -0500
Subject: [PATCH] make base.pm more strict about nonexistent module check

---
 dist/base/lib/base.pm     |  4 +++-
 dist/base/t/base.t        | 10 +++++++++-
 dist/base/t/lib/Broken.pm |  7 +++++++
 3 files changed, 19 insertions(+), 2 deletions(-)
 mode change 100644 => 100755 dist/base/t/base.t
 create mode 100644 dist/base/t/lib/Broken.pm

diff --git a/dist/base/lib/base.pm b/dist/base/lib/base.pm
index 4c4b1d2..6ed42ed 100644
--- a/dist/base/lib/base.pm
+++ b/dist/base/lib/base.pm
@@ -87,7 +87,9 @@ sub import {
                 # probably be using parent.pm, which doesn't try to
                 # guess whether require is needed or failed,
                 # see [perl #118561]
-                die if $@ && $@ !~ /^Can't locate .*? at \(eval /;
+                (my $fn = $base) =~ s!::!/!g;
+                $fn .= '.pm';
+                die if $@ && $@ !~ /^Can't locate \Q$fn\E .*? at \(eval /;
                 unless (%{"$base\::"}) {
                     require Carp;
                     local $" = " ";
diff --git a/dist/base/t/base.t b/dist/base/t/base.t
old mode 100644
new mode 100755
index 6fb24ea..4308982
--- a/dist/base/t/base.t
+++ b/dist/base/t/base.t
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 use strict;
-use Test::More tests => 11;
+use Test::More tests => 12;
 
 use_ok('base');
 
@@ -81,3 +81,11 @@ like( $@, qr/^Base class package "reallyReAlLyNotexists" is empty\./,
     eval q{ use base 'Schlozhauer' };
     ::is( $@, '', 'Can coexist with a FIELDS constant' );
 }
+
+{
+    use lib 't/lib';
+    package UsingBroken;
+    eval q{use base 'Broken';};
+    ::like( $@, qr/^Can't locate ThisModuleDoesNotExist\.pm/,
+        'base fails to compile by loading nonexistant module');
+}
diff --git a/dist/base/t/lib/Broken.pm b/dist/base/t/lib/Broken.pm
new file mode 100644
index 0000000..4d9474a
--- /dev/null
+++ b/dist/base/t/lib/Broken.pm
@@ -0,0 +1,7 @@
+package Broken;
+
+sub i_exist { 1 }
+
+eval "require ThisModuleDoesNotExist;" or die $@;
+
+1;
-- 
1.8.4.3


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Wed Dec 04 03&#8203;:05&#8203;:59 2013\, haarg wrote&#8203;:
> This is a bug report for perl from haarg@&#8203;haarg\.org\,
> generated with the help of perlbug 1\.39 running under perl 5\.18\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> 
> When attempting to load a base module\, base\.pm ignores nonexistent
> modules\.  The check it uses isn't very strict though\, and can end
> up ignoring failures that it should have caught\.  Any error message
> about failing to load a module that is in an eval will be ignored\.
> Instead\, it should only ignore the specific module that it is
> loading\.
> 
> Patch with test attached\.
> 
> 

The patch looked good to me\, at first\.

It needed s/nonexistant/nonexistent/

But when I ran 'make test' I got \*lots\* of errors&#8203;:

\#\#\#\#\#
Test Summary Report


---
uni/universal\.t                                                 \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_01\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_01\_dfs\_utf8\.t                                         \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_02\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_02\_dfs\_utf8\.t                                         \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_03\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_03\_dfs\_utf8\.t                                         \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_04\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_04\_dfs\_utf8\.t                                         \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_05\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/basic\_05\_dfs\_utf8\.t                                         \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/complex\_c3\_utf8\.t                                           \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/complex\_dfs\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/next\_inanon\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/next\_ineval\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/next\_method\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/next\_NEXT\_utf8\.t                                            \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/next\_skip\_utf8\.t                                            \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/overload\_c3\_utf8\.t                                          \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/vulcan\_c3\_utf8\.t                                            \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
mro/vulcan\_dfs\_utf8\.t                                           \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
porting/cmp\_version\.t                                           \(Wstat&#8203;: 0 Tests&#8203;: 24 Failed&#8203;: 1\)
&nbsp;  Failed test&#8203;:  15
porting/manifest\.t                                              \(Wstat&#8203;: 0 Tests&#8203;: 10219 Failed&#8203;: 3\)
&nbsp;  Failed tests&#8203;:  10214\, 10216\-10217
Files=2361\, Tests=683277\, 122 wallclock secs \(72\.89 usr  7\.48 sys \+ 437\.95 cusr 38\.01 csys = 556\.33 CPU\)
Result&#8203;: FAIL
make&#8203;: \*\*\* \[test\_harness\] Error 4
\#\#\#\#\#

Bumping the version number in base\.pm will probably resolve the failures in porting/cmp\_version\.t and porting/manifest\.t\.  But I don't have any quick explanation for the other failures\.

Can you investigate further?

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @haarg

Attached is an updated patch that fixes the test failures\, includes some additional tests\, and also fixes rt\#120687\.

**Comment by p5pRT** [comment]
### From @haarg

<details><summary>0001-make-base.pm-more-strict-about-nonexistent-module-ch.patch</summary>

```diff
From 99fde04c0b6eff8b2985b90e44efae23ed0a8f14 Mon Sep 17 00:00:00 2001
From: Graham Knop <haarg@haarg.org>
Date: Wed, 4 Dec 2013 05:54:02 -0500
Subject: [PATCH] make base.pm more strict about nonexistent module check

---
 MANIFEST                  |  1 +
 dist/base/MANIFEST        |  1 +
 dist/base/lib/base.pm     | 25 +++++++++++++++++++++++--
 dist/base/t/base.t        | 42 +++++++++++++++++++++++++++++++++++++++++-
 dist/base/t/lib/Broken.pm |  7 +++++++
 5 files changed, 73 insertions(+), 3 deletions(-)
 create mode 100644 dist/base/t/lib/Broken.pm

diff --git a/MANIFEST b/MANIFEST
index 880e830..bd7dc6b 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -2832,6 +2832,7 @@ dist/base/t/fields-5_8_0.t	See if fields work
 dist/base/t/fields-base.t	See if fields work
 dist/base/t/fields.t		See if fields work
 dist/base/t/isa.t		See if base's behaviour doesn't change
+dist/base/t/lib/Broken.pm	Test module for base.pm
 dist/base/t/lib/Dummy.pm	Test module for base.pm
 dist/base/t/lib/HasSigDie.pm	Module for testing base.pm
 dist/base/t/sigdie.t		See if base works with SIGDIE
diff --git a/dist/base/MANIFEST b/dist/base/MANIFEST
index da973e0..cdf1fdb 100644
--- a/dist/base/MANIFEST
+++ b/dist/base/MANIFEST
@@ -11,6 +11,7 @@ t/fields-5_8_0.t
 t/fields-base.t
 t/fields.t
 t/isa.t
+t/lib/Broken.pm
 t/lib/Dummy.pm
 t/lib/HasSigDie.pm
 t/sigdie.t
diff --git a/dist/base/lib/base.pm b/dist/base/lib/base.pm
index 4c4b1d2..24f7e5c 100644
--- a/dist/base/lib/base.pm
+++ b/dist/base/lib/base.pm
@@ -55,6 +55,23 @@ else {
     }
 }
 
+if ($] < 5.008) {
+    *_module_to_filename = sub {
+        (my $fn = $_[0]) =~ s!::!/!g;
+        $fn .= '.pm';
+        return $fn;
+    }
+}
+else {
+    *_module_to_filename = sub {
+        (my $fn = $_[0]) =~ s!::!/!g;
+        $fn .= '.pm';
+        utf8::encode($fn);
+        return $fn;
+    }
+}
+
+
 sub import {
     my $class = shift;
 
@@ -78,7 +95,10 @@ sub import {
             my $sigdie;
             {
                 local $SIG{__DIE__};
-                eval "require $base";
+                my $fn = _module_to_filename($base);
+                my $file = __FILE__;
+                my $line = __LINE__ + 1;
+                eval { require $fn };
                 # Only ignore "Can't locate" errors from our eval require.
                 # Other fatal errors (syntax etc) must be reported.
                 #
@@ -87,7 +107,8 @@ sub import {
                 # probably be using parent.pm, which doesn't try to
                 # guess whether require is needed or failed,
                 # see [perl #118561]
-                die if $@ && $@ !~ /^Can't locate .*? at \(eval /;
+                die if $@ && $@ !~ /^Can't locate \Q$fn\E .*? at \Q$file\E line \Q$line\E\.\n\z/s
+                          || $@ =~ /Compilation failed in require at \Q$file\E line \Q$line\E\.\n\z/;
                 unless (%{"$base\::"}) {
                     require Carp;
                     local $" = " ";
diff --git a/dist/base/t/base.t b/dist/base/t/base.t
index 6fb24ea..0bbb5be 100644
--- a/dist/base/t/base.t
+++ b/dist/base/t/base.t
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 use strict;
-use Test::More tests => 11;
+use Test::More tests => 15;
 
 use_ok('base');
 
@@ -81,3 +81,43 @@ like( $@, qr/^Base class package "reallyReAlLyNotexists" is empty\./,
     eval q{ use base 'Schlozhauer' };
     ::is( $@, '', 'Can coexist with a FIELDS constant' );
 }
+
+{
+    use lib 't/lib';
+    package UsingBroken;
+    eval q{use base 'Broken';};
+    ::like( $@, qr/^Can't locate ThisModuleDoesNotExist\.pm/,
+        'base fails to compile by loading nonexistent module');
+}
+
+SKIP: {
+    skip "unicode not supported on perl $]", 2 if $] < 5.008;
+    eval q{
+        package UsingUnicode;
+        my $base = "M\N{U+00D8}dule";
+        no strict 'refs';
+        *{"${base}::foo"} = sub {};
+        eval q{use base $base;};
+        ::is( $@, '', 'nonexistent unicode module allowed');
+    };
+
+    eval q{
+        package UsingUtf8;
+        my $base = "M\N{U+00D8}dule";
+        utf8::encode($base);
+        no strict 'refs';
+        *{"${base}::foo"} = sub {};
+        eval q{use base $base;};
+        ::is( $@, '', 'nonexistent utf8 module allowed');
+    };
+}
+
+{
+    package WithHostileINC;
+    local @INC = (@INC, "a\nb");
+    my $base = "NonExistentModule";
+    no strict 'refs';
+    *{"${base}::foo"} = sub {};
+    eval q{use base $base;};
+    ::is( $@, '', 'nonexistent module allowed when @INC has hostile entries');
+}
diff --git a/dist/base/t/lib/Broken.pm b/dist/base/t/lib/Broken.pm
new file mode 100644
index 0000000..4d9474a
--- /dev/null
+++ b/dist/base/t/lib/Broken.pm
@@ -0,0 +1,7 @@
+package Broken;
+
+sub i_exist { 1 }
+
+eval "require ThisModuleDoesNotExist;" or die $@;
+
+1;
-- 
1.8.5.1


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu Dec 05 04&#8203;:31&#8203;:45 2013\, haarg wrote&#8203;:
> Attached is an updated patch that fixes the test failures\, includes
> some additional tests\, and also fixes rt\#120687\.

This patch does pass 'make test'\.

I have pushed it for smoke\-testing to&#8203;:
smoke\-me/jkeenan/120685\-2nd\-base

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri Dec 06 18&#8203;:11&#8203;:34 2013\, jkeenan wrote&#8203;:
> On Thu Dec 05 04&#8203;:31&#8203;:45 2013\, haarg wrote&#8203;:
> > Attached is an updated patch that fixes the test failures\, includes
> > some additional tests\, and also fixes rt\#120687\.
> 
> This patch does pass 'make test'\.
> 
> I have pushed it for smoke\-testing to&#8203;:
> smoke\-me/jkeenan/120685\-2nd\-base
> 
> Thank you very much\.
> Jim Keenan
> 
> 


This branch has been smoking at&#8203;:
http&#8203;://perl\.develop\-help\.com/?b=smoke\-me%2Fjkeenan%2F120685\-2nd\-base

The branch is failing on certain smoke configurations \-\- but then those configurations seem to fail most of the time\, anyway\.  And the failures that are appearing do not appear related to base\.pm\.

I'd like to apply this patch\, but I know that there are people who\, errrm\, have opinions about base\.pm\.  I'd like to allow time for those people to sound off\.  If you have an opinion\, please speak up ASAP\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @tonycoz

On Mon Dec 09 19&#8203;:10&#8203;:34 2013\, jkeenan wrote&#8203;:
> On Fri Dec 06 18&#8203;:11&#8203;:34 2013\, jkeenan wrote&#8203;:
> > On Thu Dec 05 04&#8203;:31&#8203;:45 2013\, haarg wrote&#8203;:
> > > Attached is an updated patch that fixes the test failures\, includes
> > > some additional tests\, and also fixes rt\#120687\.
> >
> > This patch does pass 'make test'\.
> >
> > I have pushed it for smoke\-testing to&#8203;:
> > smoke\-me/jkeenan/120685\-2nd\-base
> >
> > Thank you very much\.
> > Jim Keenan
> >
> >
> 
> 
> This branch has been smoking at&#8203;:
> http&#8203;://perl\.develop\-help\.com/?b=smoke\-me%2Fjkeenan%2F120685\-2nd\-base
> 
> The branch is failing on certain smoke configurations \-\- but then
> those configurations seem to fail most of the time\, anyway\.  And the
> failures that are appearing do not appear related to base\.pm\.
> 
> I'd like to apply this patch\, but I know that there are people who\,
> errrm\, have opinions about base\.pm\.  I'd like to allow time for those
> people to sound off\.  If you have an opinion\, please speak up ASAP\.

The failures are irrelevent to the changes to base\.pm

Tony

**Comment by p5pRT** [comment]
### From @tonycoz

On Thu Dec 05 04&#8203;:31&#8203;:45 2013\, haarg wrote&#8203;:
> Attached is an updated patch that fixes the test failures\, includes
> some additional tests\, and also fixes rt\#120687\.


Thanks\, applied as 892089cd68f447dda17eb023010cd1dfb14ea0a5 with a version bump in be94f0b715e39ec6b3a6762b0bf8db5799f1c59e\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'open' to 'resolved'
